apps/src/javalab/javalabRedux.js

```diff
@@ -378,6 +384,24 @@ export const setAllEditorMetadata = (
   };
 };
 
+export const setNewFileError = error => ({
+  type: SET_NEW_FILE_ERROR,
+  error: error
+});
+
+export const clearNewFileError = () => ({
+  type: CLEAR_NEW_FILE_ERROR
+});
+
+export const setRenameFileError = error => ({
```

Comment:
is there a reason for having both `set...` and `clear...` functions? Could we just call `set...` with `null`?

Reply:
Just for clarity - we could also call `set...` with `null` and it would do the same thing.

Reply:
Should we simplify to just have `set...` then? Mainly to reduce the number of things we have to pass around/import

Reply:
Yeah, my inclination to do it this way was just that `clearError` seems a little more encapsulated than `setError(null)` because it lets the redux file handle what "clear" means. Just in case something like `setError(undefined)` yields unexpected results because we're expecting `null` somewhere else. I don't have a strong preference though so I can simplify this to reduce the number of props.

Reply:
@sanchitmalhotra126 I think you are [on the right track](https://redux.js.org/style-guide/#model-actions-as-events-not-setters) that the action describes the event (a error was cleared) and the reducer determines how that impacts the state (sets something to `null`).